apisix/plugins/kafka-logger.lua

```diff
@@ -199,15 +225,24 @@ function _M.log(conf, ctx)
     end
 
     -- reuse producer via lrucache to avoid unbalanced partitions of messages in kafka
-    local broker_list = core.table.new(core.table.nkeys(conf.broker_list), 0)
+    local length = conf.broker_list and core.table.nkeys(conf.broker_list) or #conf.brokers
```

Comment:
Just FYI: if we don't need to rewrite the content of table, we can use core.table.clone instead of `alloc + insert in loop`.

Reply:
Let me update it.

Reply:
Thanks for the remind. Improved.